src/Microsoft.Dnx.Runtime/FrameworkReferenceResolver.cs

```diff
@@ -171,8 +171,11 @@ private static FrameworkInformation GetFrameworkInformation(FrameworkName target
 
             // Skip this on mono since it has a slightly different set of reference assemblies at a different
             // location
-            if (!RuntimeEnvironmentHelper.IsMono
-                && FrameworkDefinitions.TryPopulateFrameworkFastPath(targetFramework.Identifier, targetFramework.Version, referenceAssembliesPath, out frameworkInfo))
+            // Also, if the framework we're targetting has a profile (Client profile, basically), don't use the
+            // fast path because it will be wrong.
+            if (!RuntimeEnvironmentHelper.IsMono &&
+                string.IsNullOrEmpty(targetFramework.Profile) &&
+                FrameworkDefinitions.TryPopulateFrameworkFastPath(targetFramework.Identifier, targetFramework.Version, referenceAssembliesPath, out frameworkInfo))
```

Comment:
We should pass this method the entire targetFramework and have it return false so we don't need to special case here.